🐛 Image Building has references to docker/infra

Cloud Build Logs:

```
Already have image (with digest): gcr.io/k8s-testimages/gcb-docker-gcloud:v20200619-68869a4
make: *** No rule to make target 'release-staging'.  Stop.
```

Need to remove https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/cloudbuild.yaml#L16-L25

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -13,16 +13,6 @@ steps:
     - DOCKER_BUILDKIT=1
     args:
     - release-staging
-  - name: 'gcr.io/k8s-testimages/gcb-docker-gcloud:v20200619-68869a4'
-    dir: 'test/infrastructure/docker'
-    entrypoint: make
-    env:
-    - DOCKER_CLI_EXPERIMENTAL=enabled
-    - TAG=$_GIT_TAG
-    - PULL_BASE_REF=$_PULL_BASE_REF
-    - DOCKER_BUILDKIT=1
-    args:
-    - release-staging
 substitutions:
   # _GIT_TAG will be filled with a git-based tag for the image, of the form vYYYYMMDD-hash, and
   # can be used as a substitution

